.travis.yml

```diff
@@ -23,6 +23,22 @@ jobs:
         - bazel build --cpu=stm32h7 //firmware_new/boards/frankie_v1:frankie_v1_main 
         - bazel build --cpu=stm32f4 //firmware/boards/legacy_robot_stm32f4:bin 
         - bazel build --cpu=stm32f4 //firmware/boards/legacy_dongle_stm32f4:bin 
+    - name: "Test for Memory Leaks"
+      script:
+        - cd src
+        # Test simulation, firmware and firmware_new
+        - >
+          bazel test -c dbg  \
+            --run_under="valgrind --leak-check=yes --undef-value-errors=no --error-exitcode=1"   \
+            //software/simulation/...
```

Comment:
We should probably check the simulated tests too, since they are what really run a lot of the firmware (at least at the app-level). Although we generally try avoid allocating memory in the app layer it does exist, eg. the trajectory planner.

Reply:
simulated_tests have memory leaks and I don't want to deal with it in this PR. Opened #1686 for that